src/Symfony/Component/HttpClient/Response/TraceableResponse.php

```diff
@@ -52,24 +54,36 @@ public function getHeaders(bool $throw = true): array
 
     public function getContent(bool $throw = true): string
     {
-        $this->content = $this->response->getContent(false);
-
         if ($throw) {
-            $this->checkStatusCode($this->response->getStatusCode());
+            try {
+                $this->checkStatusCode($this->response->getStatusCode());
+            } catch (HttpExceptionInterface $httpException) {
+                try {
+                    return $this->content = $this->response->getContent(false);
```

Comment:
when did we throw the caught exception?

Reply:
`HttpExceptionInterface` above? https://github.com/symfony/symfony/blob/2d7e0b02c6dd8d652fe3732d0dd29f24bda1bddc/src/Symfony/Component/HttpClient/Response/TraceableResponse.php#L203-L213
Or `TransportExceptionInterface`?
https://github.com/symfony/symfony/blob/2d7e0b02c6dd8d652fe3732d0dd29f24bda1bddc/src/Symfony/Contracts/HttpClient/ResponseInterface.php#L54
https://github.com/symfony/symfony/blob/2d7e0b02c6dd8d652fe3732d0dd29f24bda1bddc/src/Symfony/Component/HttpClient/Response/CommonResponseTrait.php#L65

Reply:
hmm, I've the feeling we swallow the exception here.

user call this method with `didthrow=true`,
then we check the status code but catch the exception
then we call getContent with throw=false

I'm maybe wrong, but it looks like when the api return a non 2xx code, we never throw it.

Reply:
the first exception should be rethrown no matter if `getContent()` throws or not. Fixed now :)